app/src/main/java/com/android/sample/ui/fridge/FridgeScreen.kt

```diff
@@ -30,29 +117,414 @@ import com.android.sample.ui.utils.PlateSwipeScaffold
+          Column(
+              modifier = Modifier.padding(PADDING_16.dp),
+              horizontalAlignment = Alignment.CenterHorizontally) {
+                Text(
+                    text =
+                        "Update Quantity: ${fridgeIngredientPair.second.name} (${fridgeIngredientPair.second.quantity})",
+                    style = MaterialTheme.typography.titleMedium,
+                    lineHeight = DIALOG_TITLE_LINE_HEIGHT.sp,
+                    fontSize = DIALOG_TITLE_FONT_SIZE.sp,
+                    color = MaterialTheme.colorScheme.onPrimary.copy(alpha = DIALOG_TITLE_ALPHA),
+                    modifier = Modifier.padding(bottom = PADDING_16.dp))
+                // Quantity Adjust Buttons
+                Row(
+                    modifier = Modifier.fillMaxWidth().padding(vertical = PADDING_16.dp),
+                    horizontalArrangement = Arrangement.Center,
+                    verticalAlignment = Alignment.CenterVertically) {
+                      Counter(
+                          count = updatedQuantity,
+                          minValue = MIN_VALUE,
+                          maxValue = MAX_VALUE,
+                          onCounterChange = { setUpdatedQuantity(it) })
+                    }
+
+                Spacer(modifier = Modifier.height(PADDING_16.dp))
+
+                // Save Button
+                Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.End) {
+                  TextButton(
+                      onClick = {
+                        setUpdatedQuantity(fridgeIngredientPair.first.quantity)
+                        hiddeEditDialog()
+                      }) {
+                        Text(
+                            text = stringResource(R.string.cancel),
+                            style = MaterialTheme.typography.bodyMedium,
+                            color = MaterialTheme.colorScheme.onSecondaryContainer,
+                        )
+                      }
+                  Spacer(modifier = Modifier.width(PADDING_8.dp))
+                  Button(
+                      onClick = {
+                        userViewModel.updateIngredientFromFridge(
+                            fridgeIngredientPair.second,
+                            updatedQuantity,
+                            fridgeIngredientPair.first.expirationDate,
+                            false)
+                        hiddeEditDialog()
+                      },
+                      colors =
+                          ButtonDefaults.buttonColors(
+                              MaterialTheme.colorScheme.onSecondaryContainer)) {
+                        Text(
+                            text = stringResource(R.string.save),
+                            style = MaterialTheme.typography.bodyMedium,
+                            color = Color.White,
+                        )
+                      }
+                }
+              }
+        }
+  }
```

Comment:
This is really cool! Just one small thing, I’m not sure if it’s the size of the button or the color, but I don’t particularly like the modify button. That said, the rest is really well done. Great job!

Reply:
I'll work on adjusting the button size next week. I chose to use the app's theme, though I understand the color might not be your preference. This button is temporary; next week, I'll link it to the EditFridgeItemScreen to allow editing the expiration date and add a direct remove button. I'm open to any UI suggestions for this screen! As mentioned in the description, I'm not fully satisfied with the current UI. My priority was to create a functional v1, with plans to refine the design next week.